### ISSUE TITLE
[restrict-plus-operands] Allowing any for summands

### ISSUE THREAD

**Issue by kachkaev** [typescript-eslint/typescript-eslint#386]
**Repro**

1. Enable the rule (see [docs](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/restrict-plus-operands.md))

2. Either
    * Create a js file where numbers or strings are added
    ```js
    export const f = (a, b) => a + b;
    ```
    * Create a ts file, where arguments of type any are added

    ```ts
    export const f = (a: any, b: any) => a + b;
    ```

3. Observe the following error in each case:
    ```
    Operands of '+' operation must either be both strings or both numbers.
    eslint(@typescript-eslint/restrict-plus-operands)
    ```

**.eslintrc.js**

```JSON
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": "error"
  }
}
```

**Expected Result**

The current result is fine, however too much strictness out of which you cannot opt out may get the rule challenging to use in existing codebases. I just tried enabling it in a monorepo with mixed js and ts code and got hundreds of errors. I understand the value behind this, just find it impossible to start using the rule to catch real issues (variable of known type `number` being added to a variable of known type `string`). Errors in js files are especially unfortunate.

Would it be possible to optionally allow `any` in the rule?

```
"@typescript-eslint/restrict-plus-operands":  ["error", { allowAny: true }]
```
or even
```
"@typescript-eslint/restrict-plus-operands":  ["error", { skipJs: true }]
"@typescript-eslint/restrict-plus-operands":  ["error", { allowAny: true, skipJs: true }]
```
Or should js files just be always skipped?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.5.0` |
| `@typescript-eslint/parser`        | `1.5.0` |
| `TypeScript`                       | `3.3.4000` |
| `ESLint`                           | `5.15.3` |
| `node`                             | `10.15.2` |
| `yarn`                              | `1.13.0` |

/cc @armano2 (author of https://github.com/typescript-eslint/typescript-eslint/pull/70)

**Comment by bradzacher** [comment]
Typescript will run over your JS code and try a little bit to give types to things based on obvious things like JSDoc comments and variable usage.
If you don't want your JS files to be linted by this rule, you can use eslint's `overrides` config to disable the rule for certain globs: https://eslint.org/docs/user-guide/configuring#disabling-rules-only-for-a-group-of-files (i.e. `"*.js").

As for skipping any's, happy for it to be added - will accept PRs for it.

Personally, I'm not sure how much value there is in that as an option, as `number + any` is a potential problem in your codebase (which this rule is attempting to warn about).

**Comment by j-f1** [comment]
Shouldn’t `any + any` be allowed though?

**Comment by bradzacher** [comment]
I would say no, not by the base rule - for the same reason that `number + any` isn't allowed.
Definitely an option to add a rule option to allow summing any (that's off by default, as it's an unsafe operation).

**Comment by papb** [comment]
Hello, I would also like to see this option, and it should be added to `restrict-template-expressions` too. It is really weird to get a suggestion to `consider using a template literal` and when changing to it the error persists :sweat_smile:
